src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -14,105 +14,95 @@ namespace System.Buffers
     public readonly partial struct ReadOnlySequence<T>
     {
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal bool TryGetBuffer(in SequencePosition start, in SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
+        private bool TryGetBuffer(in SequencePosition position, out ReadOnlyMemory<T> memory, out SequencePosition next)
         {
-            SequenceType type;
-            int startIndex = 0;
-            int endIndex = 0;
             next = default;
-            object startObject = start.GetObject();
-            if (startObject != null)
-            {
-                GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out type, out startIndex, out endIndex);
-            }
-            else
+            object positionObject = position.GetObject();
+            if (positionObject == null)
             {
-                type = SequenceType.Empty;
+                memory = default;
+                return false;
             }
 
-            int length = endIndex - startIndex;
-            object endObject = end.GetObject();
-
-            if (type != SequenceType.MultiSegment && type != SequenceType.Empty && startObject != endObject)
-                ThrowHelper.ThrowInvalidOperationException_EndPositionNotReached();
+            int endIndex = GetIndex(_sequenceEnd);
 
+            SequenceType type = GetSequenceType();
```

Comment:
Shouldn't `GetSequenceType()` take `position` as a parameter here instead of using `_sequenceStart`?

Reply:
It will 

> Shouldn't GetSequenceType() take position as a parameter here instead of using _sequenceStart?

Sequance type is stored as sing bits in _sequenceStart.GetInteger() and _sequenceEnd.GetInteger() except default (_sequenceStart.GetObject == null).

Reply:
I get that, but there is no guarantee that the SequencePosition that the user passes in would have the same sign bits as _sequenceStart.

Reply:
You can't validate the type by one sequence position. It might be `.Start` or `.End` so it might or might not have the flag.

Reply:
> I get that, but there is no guarantee that the SequencePosition that the user passes in would have the same sign bits as _sequenceStart.

We ignore position flag because it can contain any flag, from `Start` or from `End` or does not contain flag at all.
We story flags in _sequenceStart.GetInteger() and _sequenceEnd.GetInteger(), so we get flags from _sequenceStart.GetInteger() and _sequenceEnd.GetInteger().